### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,21 @@
...
+
+    for new_dir in path_to_file:
+        new_path += new_dir + "/"
+        if not os.path.exists(new_path):
+            os.mkdir(new_path)
+
+    with open(
+            f"{current_file}", "r"
+    ) as file, open(
+        f"{future_file}", "w"
```
Comment (katevasilenko) [inline app/main.py:18]: You need to put full path to file here

File: app/main.py [hunk 2]
```diff
@@ -1 +1,21 @@
...
+    path_to_file = future_file.split("/")[:-1]
+    new_path = ""
+
+    for new_dir in path_to_file:
+        new_path += new_dir + "/"
+        if not os.path.exists(new_path):
+            os.mkdir(new_path)
+
+    with open(
+            f"{current_file}", "r"
```
Comment (katevasilenko) [inline app/main.py:16]: You dont need f-string here and on 18 line

**Review by katevasilenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1 +1,25 @@
-# write your code here
+import os
+
+
+def move_file(command: str) -> None:
+    cp_command, current_file, future_file = command.split()
```
Comment (ArsenPidhoretskyi) [inline app/main.py:5]: Validate that command have exactly 3 items

File: app/main.py [hunk 4]
```diff
@@ -1 +1,25 @@
...
+    if cp_command != "mv":
+        return
+    path_to_file = os.path.split(future_file)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:8]:
```suggestion
    directories, file_name = os.path.split(future_file)
```

File: app/main.py [hunk 5]
```diff
@@ -1 +1,25 @@
...
+    new_path = ""
+
+    for new_dir in path_to_file[0].split("/"):
+        new_path += new_dir + "/"
+        if not os.path.exists(new_path):
+            os.mkdir(new_path)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:14]: You os.makedirs with exist_ok=True

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 6]
```diff
@@ -1 +1,24 @@
...
+
+def move_file(command: str) -> None:
+    cp_command, current_file, future_file = command.split()
+    if cp_command != "mv" or len(command.split()) != 3:
+        return
+    directories, file_name = os.path.split(future_file)
+    new_path = ""
+
+    for new_dir in directories.split("/"):
+        new_path += new_dir + "/"
```
Comment (katevasilenko) [inline app/main.py:12]: This part is looks useless. You split by / and that add part with /

File: app/main.py [hunk 7]
```diff
@@ -1 +1,24 @@
...
+        os.makedirs(new_path, exist_ok=True)
```
Comment (katevasilenko) [inline app/main.py:13]: Just put `directories` here and definitely out of loop

**Review by katevasilenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 8]
```diff
@@ -1 +1,24 @@
-# write your code here
+import os
+
+
+def move_file(command: str) -> None:
+    cp_command, current_file, future_file = command.split()
+    if cp_command != "mv" or len(command.split()) != 3:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:6]: Try this:
a, b, c = "hello world".split()
Because now you:
1. do the unpacking
2. and after validate

File: app/main.py [hunk 9]
```diff
@@ -1 +1,24 @@
...
+        return
+    directories, file_name = os.path.split(future_file)
+    new_path = ""
+
+    for new_dir in directories:
+        new_path += new_dir
+        os.makedirs(new_path, exist_ok=True)
```
Comment (ArsenPidhoretskyi) [inline app/main.py:13]: use os.path.join(*list_of_directoris)

File: app/main.py [hunk 10]
```diff
@@ -1 +1,24 @@
...
+
+    if directories != "":
+        with open(
+            current_file, "r"
+        ) as file, open(
+            f"{directories}/{file_name}", "w"
+        ) as new_file:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:20]: Too hard read code....
```suggestion
        with (
            open(current_file, "r") as source_file, 
            open(f"{directories}/{file_name}", "w") as destination_file
        ):
```

Comment (ArsenPidhoretskyi) [reply app/main.py:20]: BTW file, new_file is not so clear name

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by Vasyl-Smutok — CHANGES_REQUESTED** [review]
please do what mentors say

**Review by Grayder0152 — APPROVED** [review]

**Pull request closed without merging**